hcloud/load_balancers.go

```diff
@@ -44,6 +45,29 @@ func newLoadBalancers(lbOps LoadBalancerOps, ac hcops.HCloudActionClient, disabl
 	}
 }
 
+func matchNodeSelector(svc *corev1.Service, nodes []*corev1.Node) ([]*corev1.Node, error) {
+	var (
+		err           error
+		selectedNodes []*corev1.Node
+	)
+
+	selector := labels.Everything()
+	if v, ok := annotation.LBNodeSelector.StringFromService(svc); ok {
+		selector, err = labels.Parse(v)
+		if err != nil {
+			return nil, err
```

Comment:
An error here is exposed to the user as a `Event`. IMO its hard to understand with the current message where the issue actually happened, and what the user should do to fix this.

Error message from event: `Error syncing load balancer: failed to ensure load balancer: hcloud/loadBalancers.EnsureLoadBalancer: found '==', expected: !, identifier, or 'end of string'`

I would prefer if we added something along the lines of "unable to parse node-selector annotation:" to the error:

```suggestion
			return nil, fmt.Errorf("unable to parse the node-selector annotation: %w", err)
```

Reply:
Ah yeah, that makes sense :+1: just copied the snippet and haven't thought enough about it :see_no_evil: